### DISCUSSION TITLE
404 using src/app dir structure next 13

### DISCUSSION THREAD

**Discussion by kamigerami** [vercel/community#1207] [Help] [answered]
when deploying to vercel running Nextjs 13 and using the app (beta) folder structure I get a 404.

Folder structure:
```
├── src
│   ├── app
│   │   ├── head.tsx
│   │   ├── layout.tsx
│   │   └── page.tsx
.... cut ... 
...
├── tailwind.config.cjs
├── tsconfig.json
├── vercel.json
└── yarn.lock
```

I'm pretty sure it works fine when using the "pages" under src/ dir.

Any ideas?

**Comment by amyegan** [comment, marked as answer]
_Updated answer from https://github.com/vercel/community/discussions/1207#discussioncomment-4585860 below:_ The problem is happening because of an issue with the `app` directory and `i18n`. The `create-t3-app` template enables Internationalization by default, which works well for the `pages` dir but is not currently supported by the new `app` dir. You can get it to work either by removing `i18n` from your `next.config.js` to use the `app` dir without internationalization or by using only the `pages` directory to support internationalization if it is needed.

---

~~Hi @kamigerami. I wasn't able to reproduce the problem in my test app. I must be missing something in my setup. Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or public repo that I can use for debugging?~~

**Reply by kamigerami** [reply]
https://github.com/kamigerami/reproducevercelissue

**Reply by kamigerami** [reply]
Thank you for your quick response.

To reproduce:
1.  npm create t3-app@latest
2. create app dir under src/app with page.tsx layout.tsx and head.tsx
3. remove pages folder under src
4. push to vercel.

**Reply by leerob** [reply]
Hey! This is because currently there's an issue with the `app` dir and `i18n`, which that template seems to have by default. 

If you remove `i18n` from `next.config.js`, it should work. We are working on a guide for i18n in the `app` dir: https://beta.nextjs.org/docs/app-directory-roadmap#not-planned-features

**Reply by amyegan** [reply]
Thanks for sharing your repro steps @kamigerami! That was very helpful for revealing the source of the problem 😄

**Reply by kamigerami** [reply]
Hey thank you for the quick response and the workaround/solution.

**Reply by leerob** [reply]
Good news, this is now fixed on the Vercel side! It will be included in `vercel@31.1.0`.

https://github.com/vercel/vercel/pull/10243
